### ISSUE TITLE
Fork cache not working for Optimism and other networks

### ISSUE THREAD

**Issue by bernard-wagner** [NomicFoundation/hardhat#2736]
## Overview

https://github.com/NomicFoundation/hardhat/blob/e5e4a6e4d79cc79e03be90052059bb321645a542/packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts#L81

When using Hardhat to fork Optimism Mainnet or other networks that do not have a predefined `maxReorg` caching will be disabled. 

Ideally `cacheToDiskEnabled` should use `maxReorg` instead of `actualMaxReorg`. This will enable caching on any chain as long as the block number is more than `FALLBACK_MAX_REORG` behind the latest.

Alternative solution which will require more work: add an option to explicitly set the fork cache settings and if defined always cache.

## Reproduce

```
npx hardhat node --fork 'https://opt-mainnet.g.alchemy.com/v2/KEY' --fork-block-number 8540800
```

**Comment by 134dd3v** [comment]
This issue is still relevant. I'm facing this on arbitrum, the tests take about 3 mins to run and applying the changes in the PR #2737 makes it run in like 20 secs. I think this is a critical issue which affects dev ex with hardhat, and the fix is one line change. I request maintainers to re-open this issue. cc @alcuadrado @fvictorio.

**Comment by fvictorio** [comment]
You are right @134dd3v, we'll try to prioritize this.

**Comment by fvictorio** [comment]
Released in [`hardhat@2.11.2`](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.11.2)
